projects/openexr/openexr_exrheader_fuzzer.cc

```diff
@@ -0,0 +1,245 @@
+            i.name();
+            a->typeName();
+
+            if (const Box2iAttribute *ta =
+                            dynamic_cast <const Box2iAttribute *> (a))
+            {
+                ta->value();
+            }
+
+            else if (const Box2fAttribute *ta =
+                            dynamic_cast <const Box2fAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const ChannelListAttribute *ta =
+                            dynamic_cast <const ChannelListAttribute *> (a))
+            {
+                printChannelList(ta->value());
+            }
+            else if (const ChromaticitiesAttribute *ta =
+                            dynamic_cast <const ChromaticitiesAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const DoubleAttribute *ta =
+                            dynamic_cast <const DoubleAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const FloatAttribute *ta =
+                            dynamic_cast <const FloatAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const IntAttribute *ta =
+                            dynamic_cast <const IntAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const KeyCodeAttribute *ta =
+                            dynamic_cast <const KeyCodeAttribute *> (a))
+            {
+                ta->value().filmMfcCode();
+                ta->value().filmType();
+                ta->value().prefix();
+                ta->value().count();
+                ta->value().perfOffset();
+                ta->value().perfsPerFrame();
+                ta->value().perfsPerCount();
+            }
+            else if (const M33fAttribute *ta =
+                            dynamic_cast <const M33fAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const M44fAttribute *ta =
+                            dynamic_cast <const M44fAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const PreviewImageAttribute *ta =
+                            dynamic_cast <const PreviewImageAttribute *> (a))
+            {
+                ta->value().width();
+                ta->value().height();
+            }
+            else if (const StringAttribute *ta =
+                            dynamic_cast <const StringAttribute *> (a))
+            {
+                ta->value();
+            }
+            else if (const StringVectorAttribute * ta =
+                            dynamic_cast<const StringVectorAttribute *>(a))
+            {
+                for (StringVector::const_iterator i = ta->value().begin();
+                                i != ta->value().end();
+                                ++i)
+                {
+                    ;
```

Comment:
can we do something here that accesses the data in the vector?

Reply:
It doesn't look like we can do much with [`StringVectorAttribute`](https://github.com/AcademySoftwareFoundation/openexr/blob/master/OpenEXR/IlmImf/ImfStringVectorAttribute.cpp)s. The only thing the [original file](https://github.com/AcademySoftwareFoundation/openexr/blob/master/OpenEXR/exrheader/main.cpp#L473) did with this is dereference the `StringVector::const_iterator` pointer, so I've added that for now.